src/Voron/Impl/Scratch/ScratchBufferFile.cs

```diff
@@ -412,9 +412,9 @@ internal List<PageFromScratchBuffer> GetFirst10AllocatedPages()
             {
                 var pages = new List<PageFromScratchBuffer>();
 
-                foreach (var key in _parent._allocatedPages.Keys)
+                foreach (var keyValue in _parent._allocatedPages.ToArray())
```

Comment:
That is still suspect for concurrency, no?

Reply:
Correct